app/controllers/application_controller.rb

```diff
@@ -1,7 +1,7 @@
 class ApplicationController < ActionController::Base
   include ActionView::Layouts
   layout :theme_layout
-
+  before_action :require_login, except: [:new, :create]
```

Comment:
Change this because we needs to check this inside a commnet and post controller.